iamclient/iamclient.go

```diff
@@ -356,3 +418,73 @@ func (client *Client) getSystemID() (systemID string, err error) {
 
 	return response.GetSystemId(), nil
 }
+
+func (client *Client) processMessages(listener pb.IAMPublicNodesService_SubscribeNodeChangedClient) (err error) {
+	for {
+		nodeInfo, err := listener.Recv()
+		if err != nil {
+			if code, ok := status.FromError(err); ok {
+				if code.Code() == codes.Canceled {
+					log.Debug("IAM client connection closed")
+					return nil
+				}
+			}
+
+			return aoserrors.Wrap(err)
+		}
+
+		client.Lock()
+		for _, listener := range client.nodeInfoListeners {
+			listener <- pbconvert.NewNodeInfoFromPB(nodeInfo)
+		}
+		client.Unlock()
+	}
+}
+
+func (client *Client) connectPublicNodeService() {
+	go func() {
```

Comment:
Better to use `go connectPublicNodeService()` to decrease function nested level.

Reply:
done